File: apps/cloud-functions/resume_to_text_v1/logger.py
# from google.cloud import logging

# class LoggerV2(logging.Client):
#     def __init__(self, name, meta, data):
#         super(LoggerV2, self).__init__()
#         self._logger = self.logger(name)
#         self._metadata = meta
#         self._data = data
#         self._logs = []

#     def set_subprocess(self, name):
#         self._data['subProcess'] = name

#     def create_log(self, log_data, options=None):
#         if log_data.get('error'):
#             if options is None:
#                 options = {}
#             options['severity'] = logging.SeverityAlias.ERROR

#         entry_data = {**self._metadata, **(options or {})}
#         entry_payload = {**self._data, **log_data}

#         temp = self._logger.entry(entry_data, entry_payload)
#         self._logs.append(temp)

#     async def end(self):
#         if not self:
#             raise ValueError("logger not initialized")

#         await self._logger.write_entries(self._logs).catch(lambda error: print(error))

# # Usage:
# # Set your Google Cloud credentials before using this code
# # export GOOGLE_APPLICATION_CREDENTIALS="[PATH]"

# logger = None

# def init_logger(name, meta, data):
#     global logger
#     logger = LoggerV2(name, meta, data)
#     return logger

File: apps/cloud-functions/resume_socring_v3/src/logger.ts
import { Log, Logging } from "@google-cloud/logging";
import { LogEntry } from "@google-cloud/logging/build/src/entry";

// let loggerClient: Log | null = null;

type LoggerDataType = {
  timestamp?: string;
  function?: string;
  subProcess?: string;
  application_id?: string;
  company?: string;
  job_id?: string;
  status: string;
  error?: string;
};

export default class LoggerV2 extends Logging {
  private _metadata: LogEntry;
  private _data: LoggerDataType;
  private _logger: Log;
  loggerDetails: { [key: string]: string } = { application_id: "not set yet" };
  // private _logs: Entry[] = [];
  constructor(name: string, meta: LogEntry, data: LoggerDataType) {
    super();
    this._logger = super.log(name);
    this._metadata = meta;
    this._data = data;
  }
  get metadata() {
    return this._metadata;
  }
  setSupProcess(name: string) {
    this._data.subProcess = name;
  }
  async createLog(
    logData: {
      message?: string;
      subProcess?: string;
      error?: string;
    },
    options?: {
      severity?: LogEntry["severity"];
    }
  ) {
    if (logData.error) {
      options = { ...options, severity: "ERROR" };
    }
    const temp = this._logger.entry(
      { ...this._metadata, ...options },
      {
        ...this._data,
        ...logData,
      }
    );
    await this._logger.write(temp).catch((error: any) => {
      console.error(error);
    });
    // this._logs.push(temp);
  }
}
// async end() {
//   if (!this._logger) throw new Error("logger not initialized");
//   // console.log(this._logs);
//   await this._logger.write(this._logs).catch((error) => {
//     console.error(error);
//   });
// }
// }

// export const createLogger = (
//   name: string,
//   meta: LogEntry,
//   data: LoggerDataType
// ) => {
//   return new LoggerV2(name, meta, data);
// };
